### PR TITLE
Add storage interface

### PR DESCRIPTION
**Pull request by nikunjgit**

### CODE REVIEW

File: models/tag.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package models
```
Comment (benraskin92) [inline models/tag.go]: Maybe add a couple unit tests for this package? Unless you were planning on doing that in another diff.

File: models/tag.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	typeToStr := map[MatchType]string{
+		MatchEqual:     "=",
+		MatchNotEqual:  "!=",
+		MatchRegexp:    "=~",
+		MatchNotRegexp: "!~",
+	}
+	if str, ok := typeToStr[m]; ok {
+		return str
+	}
+	panic("unknown match type")
```
Comment (benraskin92) [inline models/tag.go]: Should we `panic` or return an error here?

Comment (nikunjgit) [reply models/tag.go]: String() can't really return an error. Also, this shouldn't have any other value.

File: ts/series.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"context"
+	"time"
+)
+
+// A Series is the public interface to a block of timeseries values.  Each block has a start time,
+// a logical number of steps, and a step size indicating the number of milliseconds represented by each point.
+type Series struct {
+	name      string
```
[the remark this answers is not in the document]
Comment (nikunjgit) [reply ts/series.go]: Added a method

File: storage/interface.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+	"github.com/m3db/m3coordinator/models"
+	"github.com/m3db/m3coordinator/ts"
+
+	xtime "github.com/m3db/m3x/time"
+)
+
+// Storage provides an interface for reading and writing to the tsdb
+type Storage interface {
+	Queryable
```
[the remark this answers is not in the document]
Comment (nikunjgit) [reply storage/interface.go]: done

**Pull request merged**